src/app/services/baw-api/api-common.ts

```diff
@@ -46,6 +46,30 @@ export function option(x?: New | Filter | Empty) {
   return x ? x : Empty;
 }
 
+/**
+ * Allow filters while accepting a model id to refine search
+ * @param filters Base Filters
+ * @param key Key to match model id against
+ * @param model Additional filter by model id
+ */
+export function filterById<T extends AbstractModel>(
+  filters: Filters<T>,
+  key: keyof T,
+  model: IdOr<AbstractModel>
+): Filters<T> {
+  const { filter, ...meta } = filters;
+
+  return {
+    ...meta,
+    filter: {
+      ...filter,
+      [key]: {
+        eq: model instanceof AbstractModel ? model.id : model,
+      },
+    },
+  };
+}
+
```

Comment:
This method is redundant for `service<T>` and `Model<T>`. You don't need `/filter` to get a specific object of the same type... you just use `/:id`

It is useful for `service<T>` and `Model<U>` - i.e. some other model that can be queried that is associated with `T`.

Add documentation to that effect maybe?

Reply:
```typescript
/**
 * Modify a base filter to add a foreign key condition
 * @param filters Base Filters
 * @param key Foreign key
 * @param model Foreign key value
 */
export function filterByForeignId<T extends AbstractModel>(
```